tests/test_loadDiaCatalogs.py

```diff
@@ -64,19 +64,23 @@ def setUp(self):
         self.addCleanup(os.close, self.db_file_fd)
 
         self.apdbConfig = ApdbSql.init_database(db_url="sqlite:///" + self.db_file)
+        self.config_file = tempfile.NamedTemporaryFile()
+        self.addCleanup(self.config_file.close)
+        self.apdbConfig.save(self.config_file.name)
         self.apdb = Apdb.from_config(self.apdbConfig)
         self.schema = readSchemaFromApdb(self.apdb)
 
         self.exposure = makeExposure(False, False)
+        self.regionTime = makeRegionTime(exposure=self.exposure)
 
         self.diaObjects = makeDiaObjects(20, self.exposure, rng)
         self.diaSources = makeDiaSources(
             100, self.diaObjects["diaObjectId"].to_numpy(), self.exposure, rng)
         self.diaForcedSources = makeDiaForcedSources(
             200, self.diaObjects["diaObjectId"].to_numpy(), self.exposure, rng)
 
-        self.dateTime = self.exposure.visitInfo.date
-        self.apdb.store(self.dateTime.toAstropy(),
+        self.dateTime = self.regionTime.timespan.begin.tai
+        self.apdb.store(self.dateTime,
```

Comment:
I am inclined to leave this mostly as-is, since it is just setting up an APDB with fake objects and sources for the tests to load from. I'll make it more clear that these are simulated prior observations by setting the date here a month prior to the test exposure. My understanding of `apdb.getDiaSources` is that it is intended to retrieve all diaSources with a time equal to or less than the supplied time, and equal to or greater than 12*30 days prior to the supplied time.

Reply:
Yes, but your fake sources have the _exact same timestamp_ that the implementation of `LoadDiaCatalogsTask` happens to search for. That could lead to all sorts of false negatives for bugs that only appear when the times are different (which is what will happen in real use).